nixos/modules/services/web-apps/cryptpad.nix

```diff
@@ -0,0 +1,213 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.cryptpad;
+
+  inherit (lib) strings mdDoc mkOption types mkIf mkMerge;
+
+  # The Cryptpad configuration file isn't JSON, but a JavaScript source file that assigns a JSON value
+  # to a variable.
+  cryptpadConfigFile = builtins.toFile "cryptpad_config.js" ''
+    module.exports = ${builtins.toJSON cfg.config}
+  '';
+
+  # Derive domain names for Nginx configuration from Cryptpad configuration
+  mainDomain = strings.removePrefix "https://" cfg.config.httpUnsafeOrigin;
+  sandboxDomain = if isNull cfg.config.httpSafeOrigin then mainDomain else strings.removePrefix "https://" cfg.config.httpSafeOrigin;
+
+in {
+  options.services.cryptpad = {
+    enable = lib.mkEnableOption "cryptpad";
+
+    package = lib.mkPackageOption pkgs "cryptpad" {};
+
+    configureNginx = mkOption {
+      description = mdDoc ''
+        Configure Nginx as a reverse proxy for Cryptpad.
+        Note that this makes some assumptions on your setup, and sets settings that will
+        affect other virtualHosts running on your Nginx instance, if any.
+        Alternatively you can configure a reverse-proxy of your choice.
+      '';
+      type = types.bool;
+      default = false;
+    };
+
+    confinement = mkOption {
+      type = types.bool;
+      default = false;
+      description = mdDoc ''
+        FIXME: Enables 'confinement' (explain what it does?)
+        Enable the nixos systemd.services.cryptpad.confinement setting, including the necessary
+        extra bind mounts.
+      '';
+    };
+
+    config = mkOption {
+      description = mdDoc ''
+        Cryptpad configuration settings.
+        See https://github.com/cryptpad/cryptpad/blob/main/config/config.example.js for a more extensive
+        reference documentation.
+        Test your deployed instance through `https://<domain>/checkup/`.
+      '';
+      type = types.submodule {
+        freeformType = (pkgs.formats.json {}).type;
+        options = {
+          httpUnsafeOrigin = mkOption {
+            type = types.str;
+            example = "https://cryptpad.example.com";
+            default = "";
+            description = mdDoc "This is the URL that users will enter to load your instance";
+          };
+          httpSafeOrigin = mkOption {
+            type = types.nullOr types.str;
+            example = "https://cryptpad-ui.example.com. Apparently optional but recommended.";
```

Comment:
huh that Apparently optional message was wrong in my commit, but probably belongs to description more than the example...

Reply:
Ah I realize that now too. I missed that earlier.